PathTracking/rear_wheel_feedback/rear_wheel_feedback.py

```diff
@@ -26,34 +22,79 @@
 L = 2.9  # [m]
 
 show_animation = True
-#  show_animation = False
-
 
 class State:
-
     def __init__(self, x=0.0, y=0.0, yaw=0.0, v=0.0):
         self.x = x
         self.y = y
         self.yaw = yaw
         self.v = v
 
-
-def update(state, a, delta):
-
-    state.x = state.x + state.v * math.cos(state.yaw) * dt
-    state.y = state.y + state.v * math.sin(state.yaw) * dt
-    state.yaw = state.yaw + state.v / L * math.tan(delta) * dt
-    state.v = state.v + a * dt
-
-    return state
-
+    def update(self, a, delta, dt):
+        self.x   = self.x + self.v * math.cos(self.yaw) * dt
+        self.y   = self.y + self.v * math.sin(self.yaw) * dt
+        self.yaw = self.yaw + self.v / L * math.tan(delta) * dt
+        self.v   = self.v + a * dt
+
+class TrackSpline:
+    def __init__(self, x, y):
+        x, y = map(np.asarray, (x, y))
+        s = np.append([0],(np.cumsum(np.diff(x)**2) + np.cumsum(np.diff(y)**2))**0.5)
+
+        self.X = interpolate.CubicSpline(s, x)
+        self.Y = interpolate.CubicSpline(s, y)
+
+        self.dX = self.X.derivative(1)
+        self.ddX = self.X.derivative(2)
+
+        self.dY = self.Y.derivative(1)
+        self.ddY = self.Y.derivative(2)
+
+        self.length = s[-1]
+    
+    def yaw(self, s):
```

Comment:
calc_yaw is better? function should be start by a verb.

Reply:
renamed